src/lucky/memoizable.cr

```diff
@@ -15,25 +15,69 @@ module Lucky::Memoizable
   # then each subsequent call returns the user record.
   #
   # The `memoize` method will raise a compile time exception if you forget to include
-  # a return type for your method, or if your return type is a `Union`.
-  # Arguments are not allowed in memoized methods because these can change the return value.
+  # a return type for your method, or if any arguments are missing a type.
+  # Arguments are allowed but as soon as one changes, the previous value is no longer held on to.
```

Comment:
I'm not totally sure I understand this part. I think it does hold onto the value, but it creates a new value for the new arguments?

I wonder if this might just be expected and so this sentence could be left off. What do you think?

Reply:
I can't tell if we're on the same page or not.  Calling the memoized method with new arguments and then with the old arguments will re-run the memoized method. Does the updated description make this more clear?

Reply:
Ohhhh, so if you do `my_method(1)` and then `my_method(2)`. `my_method(2)` will *overwrite* the cached value. So if you do `my_method(1)` again it will re-run the method.

Is that right? If so I think I got it and the doc changes look good!

Reply:
Exactly 👍